FinalProject/StudentTests.java

```diff
@@ -574,6 +600,40 @@ void ratingByKeywordTest2() {
     assertEquals(v1, v2);
   }
 
+  // check for abundant words in one sentence
+  @Test
+  @DisplayName("RatingByKeyword Test 3")
```

Comment:
The data format here is wrong, can you change this test to this?
```java
  @Test
  @DisplayName("RatingByKeyword Test 3")
  void ratingByKeywordTest3() {
    Parser parser1 = new Parser("");
    Parser parser2 = new Parser("");
    ArrayList<String[]> dataArray1 = new ArrayList<>();
    ArrayList<String[]> dataArray2 = new ArrayList<>();

    HashMap<String, Integer> fields = new HashMap<>();
    fields.put("professor_name", 0);
    fields.put("school_name", 1);
    fields.put("department_name", 2);
    fields.put("post_date", 3);
    fields.put("student_star", 4);
    fields.put("student_difficult", 5);
    fields.put("comments", 6);
    fields.put("gender", 7);

    parser1.fields = fields;
    parser2.fields = fields;

    String[] arr1 = {"Diana  Oqimachi", "Long Beach City College", "Counseling department", "08/26/2014", "5", "1", "fun"};
    String[] arr2 = {"Diana  Oqimachi", "Long Beach City College", "Counseling department", "08/26/2014", "5", "1", "Fun fun fun!"};
    dataArray1.add(arr1);
    dataArray2.add(arr2);

    parser1.data = dataArray1;
    parser2.data = dataArray2;

    DataAnalyzer analyzer1 = new RatingByKeyword(parser1);
    DataAnalyzer analyzer2 = new RatingByKeyword(parser2);

    MyHashTable<String, Integer> output1 = analyzer1.getDistByKeyword("fun");
    MyHashTable<String, Integer> output2 = analyzer2.getDistByKeyword("fun");

    ArrayList<String> k1 = output1.getKeySet();
    ArrayList<String> k2 = output2.getKeySet();
    ArrayList<Integer> v1 = output1.getValueSet();
    ArrayList<Integer> v2 = output2.getValueSet();

    Collections.sort(k1);
    Collections.sort(k2);
    Collections.sort(v1);
    Collections.sort(v1);

    assertEquals(k1, k2);
    assertEquals(v1, v2);
  }
```
I just corrected the data format and added fields as well